File: sem3/map/ToyLanguageInterpreter/src/view/commands/ExecuteOneStepCommand.java
package view.commands;

import controller.Controller;
import utils.exceptions.InterpreterException;

import java.util.Scanner;

/**
 * Created by mirko on 14/11/2016.
 */
public class ExecuteOneStepCommand extends Command {
    private Controller controller;

    public ExecuteOneStepCommand(String key, String description, Controller controller) {
        super(key, description);
        this.controller = controller;
    }

    @Override
    public void execute(Scanner scanner) {
        try {
            controller.executeOneStep(controller.getCurrentProgram());
            System.out.println(controller.currentProgramToString());
            System.out.flush();
        } catch (InterpreterException error) {
            System.out.println(error.toString());
            System.out.flush();
        }
    }
}

File: sem3/map/ToySharpLanguageInterpreter/ToySharpLanguageInterpreter/src/view/commands/ExecuteOneStepCommand.cs
using System;
using System.IO;

namespace ToySharpLanguageInterpreter
{
	public class ExecuteOneStepCommand : Command
	{
		private Controller controller;

		public ExecuteOneStepCommand(string key, string description, Controller controller) : base(key, description)
		{
			this.controller = controller;
		}


		public override void execute()
		{
			try
			{
				controller.executeOneStep(controller.getCurrentProgram());
				Console.WriteLine(controller.currentProgramToString());
			}
			catch (InterpreterException error)
			{
				Console.WriteLine(error.Message);
			}
		}
	}
}